src/sso-mib-tool.c

```diff
@@ -179,6 +226,112 @@ static void print_prt_token(MIBPrt *token, int decode)
 	print_account(mib_prt_get_account(token), "# ");
 }
 
+static void json_builder_add_jwt_token(JsonBuilder *builder, const gchar *token)
+{
+	char *grants = NULL;
+	char *hdrs = NULL;
+	if (decode_headers_and_claims(token, &grants, &hdrs) != 0) {
+		g_print("Error: Failed to decode JWT\n");
+		return;
+	}
+
+	json_builder_begin_object(builder);
+
+	GError *error = NULL;
+	JsonParser *parser = json_parser_new();
+	if (json_parser_load_from_data(parser, hdrs, -1, &error)) {
+		JsonNode *node = json_parser_get_root(parser);
+		json_builder_set_member_name(builder, "headers");
+		JsonNode *copied = json_node_copy(node);
+		json_builder_add_value(builder, copied);
+	} else {
+		g_printerr("Error parsing JSON string: %s\n", error->message);
+		g_error_free(error);
+	}
+
+	error = NULL;
+	if (json_parser_load_from_data(parser, grants, -1, &error)) {
+		JsonNode *node = json_parser_get_root(parser);
+		json_builder_set_member_name(builder, "grants");
+		JsonNode *copied = json_node_copy(node);
+		json_builder_add_value(builder, copied);
+	} else {
+		g_printerr("Error parsing JSON string: %s\n", error->message);
+		g_error_free(error);
```

Comment:
No need for that. Just call `g_clear_error()` when freeing the other resources (below).